Bottom: db8b25c7e9ee8816c8ad3987a22055812fbd5967
Top:    b9a17a705b96a6a3d9d671fa95a4bfaf74cb5be4
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 15:42:07 +0000

Disable texturing the board's soldermask


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 66837da..08d70eb 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -380,6 +380,7 @@ ghid_draw_grid (hidGC gc, BoxType *drawn_area)
   glEnable (GL_STENCIL_TEST);
 }
 
+#if 0
 /* XXX: Refactor this into hidgl common routines */
 static void
 load_texture_from_png (char *filename)
@@ -420,6 +421,7 @@ load_texture_from_png (char *filename)
 
   g_object_unref (pixbuf);
 }
+#endif
 
 static void
 ghid_draw_bg_image (void)
@@ -1513,8 +1515,8 @@ static void
 GhidDrawMask (int side, BoxType * screen)
 {
   render_priv *priv = gport->render_priv;
-  static bool first_run = true;
-  static GLuint texture;
+//  static bool first_run = true;
+//  static GLuint texture;
   int thin = TEST_FLAG(THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB);
   LayerType *Layer = LAYER_PTR (side == TOP_SIDE ? top_soldermask_layer : bottom_soldermask_layer);
   struct poly_info info;
@@ -1549,6 +1551,7 @@ GhidDrawMask (int side, BoxType * screen)
   hid_draw_set_color (out->fgGC, PCB->MaskColor);
   ghid_set_alpha_mult (out->fgGC, thin ? 0.35 : 1.0);
 
+#if 0
   if (first_run) {
     glGenTextures (1, &texture);
     glBindTexture (GL_TEXTURE_2D, texture);
@@ -1576,6 +1579,7 @@ GhidDrawMask (int side, BoxType * screen)
   glTexParameterf (GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);
   glTexParameterf (GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_REPEAT);
   glEnable (GL_TEXTURE_2D);
+#endif
 
   if (!PCB->Data->outline_valid) {
 
@@ -1596,15 +1600,17 @@ GhidDrawMask (int side, BoxType * screen)
   poly_FreeContours (&polygon.NoHoles);
   ghid_set_alpha_mult (out->fgGC, 1.0);
   hidgl_flush_triangles (priv->hidgl);
+#if 0
   glDisable (GL_TEXTURE_GEN_S);
   glDisable (GL_TEXTURE_GEN_T);
   glBindTexture (GL_TEXTURE_2D, 0);
   glDisable (GL_TEXTURE_2D);
+#endif
   hidgl_shader_activate (circular_program);
 
   hid_draw_use_mask (&ghid_graphics, HID_MASK_OFF);
 
-  first_run = false;
+//  first_run = false;
 }
 
 static int
